fabtests/multinode/src/core_coll.c

```diff
@@ -241,7 +241,7 @@ static int sum_all_reduce_test_run()
 	if (err) {
 		FT_DEBUG("SUM AllReduce collective not supported: %d (%s)\n",
 			 err, fi_strerror(err));
-		return err;
+		return 0;
```

Comment:
rather than treating them as passed, maybe we should check the return value for `-FI_ENOSYS` in the main loop and report as 'skipped' or 'not supported' instead.

Reply:
OK - I propose FI_EPERM as it is not used by collective providers at all.

Reply:
I believe `-FI_ENOSYS` should be sufficient, as it should be reported as the error if the collective is not implemented

Reply:
But it is possible that `-FI_ENOSYS` can be returned by a collective operation itself.

Reply:
I think it is sufficient to keep the return value as it is and check for `-FI_ENOSYS` at the main loop (L624~L625).  The debug output will help identify if the error is from the query or the operation.

Reply:
@j-xiong, what shall be the final result of `multinode_run_tests()` if some steps are skipped?
If we have a collectives operation provider that does not support a particular operation we should not test it at all.

Reply:
The FT_DEBUG messages throughout this file can be converted to FT_PRINTERR() calls.  That's sufficient to show where an error occurs.  The places where "not supported" is printed may not be correct, as we could have hit an actual error.  There's no checks that the error code was actually ENOSYS.  So, the changes now ignore actual errors.

If query() did return success that a collective is supported, we shouldn't get ENOSYS later on.

Either multinode_run_tests() should check for ENOSYS and continue if that's the case, or we need to check for ENOSYS on all of the query calls and return success back to multinode_run_tests() (like you did here).  My preference is to isolate that check into multinode_run_tests().

Reply:
I have separated multinode tests' changes into a separate #8295 PR.